packages/@aws-cdk/aws-lambda-nodejs/lib/esbuild-bundler.ts

```diff
@@ -0,0 +1,243 @@
+      return false;
+    }
+  }
+
+  public static clearRunsLocallyCache(): void { // for tests
+    this._runsLocally = undefined;
+  }
+
+  /** Cache for local run check */
+  private static _runsLocally?: boolean;
+
+  /** Cache for lock file */
+  private static lockFile?: LockFile;
+
+  // Core bundling options
+  public readonly image: cdk.BundlingDockerImage;
+  public readonly command: string[];
+  public readonly environment?: { [key: string]: string };
+  public readonly local?: cdk.ILocalBundling;
+
+  private relativeEntryPath: string;
+
+  constructor(private readonly props: EsBuildBundlerProps) {
+    this.relativeEntryPath = path.relative(props.projectRoot, path.resolve(props.entry));
+
+    const shouldBuildImage = props.forceDockerBundling || !EsBuildBundler.runsLocally;
+    this.image = shouldBuildImage
+      ? props.bundlingDockerImage ?? cdk.BundlingDockerImage.fromAsset(path.join(__dirname, '../lib'), {
+        buildArgs: {
+          ...props.buildArgs ?? {},
+          IMAGE: props.runtime.bundlingDockerImage.image,
+          ESBUILD_VERSION: props.esbuildVersion ?? ESBUILD_VERSION,
+        },
+      })
+      : cdk.BundlingDockerImage.fromRegistry('dummy'); // Do not build if we don't need to
+
+    const bundlingCommand = this.createBundlingCommand(cdk.AssetStaging.BUNDLING_INPUT_DIR, cdk.AssetStaging.BUNDLING_OUTPUT_DIR);
+
+    this.command = ['bash', '-c', bundlingCommand];
+    this.environment = props.environment;
+
+    // Define local bundling if Docker is not forced
+    if (!props.forceDockerBundling) {
+      const osPlatform = os.platform();
+      const createLocalCommand = (outputDir: string) => this.createBundlingCommand(props.projectRoot, outputDir, osPlatform);
+
+      this.local = {
+        tryBundle(outputDir: string) {
+          if (!EsBuildBundler.runsLocally) {
+            return false;
+          }
+
+          const localCommand = createLocalCommand(outputDir);
+
+          exec(
+            osPlatform === 'win32' ? 'cmd' : 'bash',
+            [
+              osPlatform === 'win32' ? '/c' : '-c',
+              localCommand,
+            ],
+            {
+              env: { ...process.env, ...props.environment ?? {} },
+              stdio: [ // show output
+                'ignore', // ignore stdio
+                process.stderr, // redirect stdout to stderr
+                'inherit', // inherit stderr
+              ],
+              cwd: path.dirname(props.entry),
+            });
+
+          return true;
+        },
+      };
+    }
+  }
+
+  public createBundlingCommand(inputDir: string, outputDir: string, osPlatform: NodeJS.Platform = 'linux'): string {
+    const pathJoin = osPathJoin(osPlatform);
+
+    // Mark the modules that we are going to install as externals also
+    const externals = [...this.props.externals ?? [], ...this.props.nodeModules ?? []];
+
+    const esbuildCommand: string = [
+      'npx', 'esbuild',
```

Comment:
it would also be good to proxy `esbuild`'s `loader` option, there is one obvious use case, in our project we use graphql and schema is defined in `.graphql` files, so to bundle this file I use `loader: { '.graphql': 'text' }`